### PR TITLE
make periods_per_year a standardized constant, and correctly implement it for cost calcs

### PR DESCRIPTION
**Pull request by quant9**
My understanding that making the optimizer period-agnostic is on the roadmap, but in the meantime I think it makes sense to support user specified periods_per_year (e.g., 250, 252, and 365). I've made a small change to let one set periods_per_year in the simulator and also corrects the flow-through to the cost calculations.

I'd note that 252 is hard-coded into various tests & other areas in the codebase and it may make sense to make it a "global" constant.

### CODE REVIEW

File: cvxportfolio/simulator.py [hunk 1]
```diff
@@ -147,10 +148,16 @@ class MarketData:
...
-        **kwargs):
+        returns=None,
+        volumes=None,
+        prices=None, 
+        cash_key='USDOLLAR',
+        base_location=BASE_LOCATION, 
+        periods_per_year=PPY,
+        min_history=PPY,
+        max_contiguous_missing='10d',
+        **kwargs,
```
Comment (quant9) [inline cvxportfolio/simulator.py:159]: I note that these kwargs, as well as the ones in the cost functions, are not currently used.

Comment (enzbus) [reply cvxportfolio/simulator.py:159]: They are, you can override them by passing them to MarketSimulator (it all passes the same `**kwargs`).

**Comment by enzbus** [comment]
Ok, I'll merge this, but the plan is to have periods_per_year discovered automatically (not sure if I'll do it this weekend). Also PPY appears in other modules, but it should be fine for now.

**Comment by enzbus** [comment]
I made it obtained automatically from `MarketData.returns.index` does it look good?

**Comment by quant9** [comment]
> I made it obtained automatically from `MarketData.returns.index` does it look good?

sorry, is this a question for me / what is this referring to?

**Comment by enzbus** [comment]
I changed your PR with this commit: https://github.com/cvxgrp/cvxportfolio/commit/5655bdd72cb743558b894aaaa88241e54f11e877 
From here it should be easy to resample (monthly, ...) in MarketData

**Pull request merged**